test/metabase/automagic_dashboards/core_test.clj

```diff
@@ -1598,3 +1598,69 @@
               (is (some? base-data))
               (is (some? filtered-data))
               (is (not= base-data filtered-data)))))))))
+
+(deftest compare-to-the-rest-with-expression-16680-test
+  (testing "Ensure a valid comparison dashboard is generated with custom expressions (fixes 16680)"
+    (mt/dataset sample-dataset
+      (mt/with-test-user :rasta
+        (let [left                 (query/adhoc-query
+                                     {:database (mt/id)
+                                      :type     :query
+                                      :query
+                                      {:source-table (mt/id :orders)
+                                       :expressions  {"TestColumn" [:+ 1 1]}
+                                       :aggregation  [[:count]],
+                                       :breakout     [[:expression "TestColumn"]
+                                                      [:field (mt/id :orders :created_at)
+                                                       {:base-type :type/DateTime :temporal-unit :month}]]}})
```

Comment:
is `:base-type` necessary here? I would expect just the temporal unit is necessary

Reply:
No, removed from test.